src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
+        derivative[0][1] =  - beta*3.0 *pi/2.0 * std::cos(pi/2.0*x_ref[0])*std::sin(3.0*pi/2.0*x_ref[1]);
+
+        derivative[1][0] =  beta*2.0*pi*std::cos(2.0*pi*(x_ref[0]))*std::cos(pi/2.0*x_ref[1]);
+        derivative[1][1] =  1.0 -beta*pi/2.0*std::sin(2.0*pi*(x_ref[0]))*std::sin(pi/2.0*x_ref[1]);  
+
+    //J. Chan nonsym
+
+//        derivative[0][0] =1.0 + beta* pi*4.0 * std::cos(pi*4.0 * (x_ref[0])) * std::sin(pi * (x_ref[1]+1.0));
+//        derivative[0][1] =      beta* pi * std::sin(pi*4.0 * (x_ref[0])) * std::cos(pi * (x_ref[1]+1.0));
+
+//        derivative[1][0] =    -  beta*pi *2.0 * 0.5*std::cos(pi*2.0 * (x_ref[0])) * std::sin(pi /2.0 * (x_ref[1]+1.0));
+//        derivative[1][1] =1.0 -  beta*pi /2.0 * 0.5*std::sin(pi*2.0 * (x_ref[0])) * std::cos(pi /2.0 * (x_ref[1]+1.0));
+
+        //Gassner
+//        derivative[0][0] = 1.0;        
+//        derivative[0][1] = - 2.0 * pi * 0.1*std::cos(2.0 * pi * x_ref[1]);
+                          
+//        derivative[1][0] =   2.0 * pi * 0.1*std::cos(2.0 * pi * x_ref[0]);
+//        derivative[1][1] = 1.0;
+    }
+    else if(dim==3){
+        derivative[0][0] = 1.0 + alpha*pi*std::cos(pi*x_ref[0])*std::sin(pi*x_ref[1]);
+        derivative[0][1] =  alpha*pi*std::sin(pi*x_ref[0])*std::cos(pi*x_ref[1]);
+        derivative[0][2] = 0.0; 
+
+        derivative[1][0] =       alpha*pi*exp(1.0-x_ref[1])*std::cos(pi*x_ref[0])*std::sin(pi*x_ref[1]);
+        derivative[1][1] =  1.0 -alpha*exp(1.0-x_ref[1])*(std::sin(pi*x_ref[0])+std::sin(pi*x_ref[1])) +alpha*exp(1.0-x_ref[1])*(std::sin(pi*x_ref[0])+std::cos(pi*x_ref[1]));  
+        derivative[1][2] =  0.0;
+        double xtemp = x_ref[0] + alpha*std::sin(pi*x_ref[0])*std::sin(pi*x_ref[1]); 
+        double ytemp = x_ref[1] + alpha*exp(1.0-x_ref[1])*std::sin(pi*x_ref[0])*std::sin(pi*x_ref[1]); 
+        derivative[2][0] = 1.0/10.0*pi*(std::cos(2.0*pi*xtemp)*derivative[0][0] + std::cos(2.0*pi*ytemp)*derivative[1][0]);
+        derivative[2][1] = 1.0/10.0*pi*(std::cos(2.0*pi*xtemp)*derivative[0][1] + std::cos(2.0*pi*ytemp)*derivative[1][1]);
+        derivative[2][2] = 1.0;
+    }
+
+        dealii::FullMatrix<double> Jacobian_inv(dim);
+        Jacobian_inv.invert(derivative);
+        dealii::Vector<double> Newton_Step(dim);
+        Jacobian_inv.vmult(Newton_Step, function);
+        for(int idim=0; idim<dim; idim++){
+            x_ref[idim] -= Newton_Step[idim];
+        }
+        flag=0;
+        for(int idim=0; idim<dim; idim++){
+            if(std::abs(function[idim]) < 1e-15)
+                flag++;
+        }
+        if(flag == dim)
+            break;
+    }
+    std::vector<double> function_check(dim);
+    if(dim==2){
+        function_check[0] = x_ref[0] + beta*std::cos(pi/2.0*x_ref[0])*std::cos(3.0*pi/2.0*x_ref[1]);
+        function_check[1] = x_ref[1] + beta*std::sin(2.0*pi*(x_ref[0]))*std::cos(pi/2.0*x_ref[1]);
+    //J.Chan nonsym
```

Comment:
add a parameter to run these commented ones?

Reply:
fixed